modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/PrecisionAtN.java

```diff
@@ -194,9 +196,69 @@ public final boolean equals(Object obj) {
         PrecisionAtN other = (PrecisionAtN) obj;
         return Objects.equals(n, other.n);
     }
-    
+
     @Override
     public final int hashCode() {
         return Objects.hash(n);
     }
+
+    public static class Breakdown implements MetricsBreakdown {
```

Comment:
I imagine this kind of information will be part of the "detailed_hits" section (to be added later, name pending).